src/cmd/flux-exec.c

```diff
@@ -244,10 +245,69 @@ static void stdin_cb (flux_reactor_t *r, flux_watcher_t *w,
     }
 }
 
-static void signal_cb (int signum)
+static void kill_completion_cb (flux_subprocess_t *p)
+{
+    flux_subprocess_destroy (p);
+}
+
+static flux_subprocess_t *imp_kill (flux_subprocess_t *p, int signum)
+{
+    flux_cmd_t *cmd;
+    flux_subprocess_ops_t ops = {
+        .on_completion = kill_completion_cb,
+        .on_state_change = NULL,
+        .on_channel_out = NULL,
+        .on_stdout = output_cb,
+        .on_stderr = output_cb,
+    };
+
+    pid_t pid = flux_subprocess_pid (p);
+    int rank = flux_subprocess_rank (p);
+
+    if (!(cmd = flux_cmd_create (0, NULL, environ))
+        || flux_cmd_argv_append (cmd, imp_path) < 0
+        || flux_cmd_argv_append (cmd, "kill") < 0
+        || flux_cmd_argv_appendf (cmd, "%d", signum) < 0
+        || flux_cmd_argv_appendf (cmd, "-%ld", (long) pid)) {
```

Comment:
For consistency the last `flux_cmd_argv_appendf()` call in the chain should probably check for < 0